course6/toyapp/displaylib/sdl.h

```diff
@@ -0,0 +1,4 @@
+#pragma once
+
+#define SDL_MAIN_HANDLED
```

Comment:
This looks like a hack. Why is it needed?
I think it's needed because you use "main" without parameters. This is OK with "vanilla" main, which gets special treatment by the compiler, but once you include SDL_main.h, SDL will define a macro called "main", which resolves to ```SDL_main(int, char**)```. This maycause compilation failure if you use it without ```(int, char**)``` params.

Reply:
Thanks, I will check out SDL main!